map_claims.go

```diff
@@ -2,150 +2,88 @@ package jwt
-		}
-	}
-	return verifyAud(aud, cmp, req)
+// GetExpirationTime implements the Claims interface.
+func (m MapClaims) GetExpirationTime() *NumericDate {
+	return m.ParseNumericDate("exp")
 }
 
-// VerifyExpiresAt compares the exp claim against cmp (cmp <= exp).
-// If req is false, it will return true, if exp is unset.
-func (m MapClaims) VerifyExpiresAt(cmp int64, req bool) bool {
-	cmpTime := time.Unix(cmp, 0)
-
-	v, ok := m["exp"]
-	if !ok {
-		return !req
-	}
-
-	switch exp := v.(type) {
-	case float64:
-		if exp == 0 {
-			return verifyExp(nil, cmpTime, req)
-		}
-
-		return verifyExp(&newNumericDateFromSeconds(exp).Time, cmpTime, req)
-	case json.Number:
-		v, _ := exp.Float64()
+// GetNotBefore implements the Claims interface.
+func (m MapClaims) GetNotBefore() *NumericDate {
+	return m.ParseNumericDate("nbf")
+}
 
-		return verifyExp(&newNumericDateFromSeconds(v).Time, cmpTime, req)
-	}
+// GetIssuedAt implements the Claims interface.
+func (m MapClaims) GetIssuedAt() *NumericDate {
+	return m.ParseNumericDate("iat")
+}
 
-	return false
+// GetAudience implements the Claims interface.
+func (m MapClaims) GetAudience() ClaimStrings {
+	return m.ParseClaimsString("aud")
 }
 
-// VerifyIssuedAt compares the exp claim against cmp (cmp >= iat).
-// If req is false, it will return true, if iat is unset.
-func (m MapClaims) VerifyIssuedAt(cmp int64, req bool) bool {
-	cmpTime := time.Unix(cmp, 0)
+// GetIssuer implements the Claims interface.
+func (m MapClaims) GetIssuer() string {
+	return m.ParseString("iss")
+}
 
-	v, ok := m["iat"]
+// ParseNumericDate tries to parse a key in the map claims type as a number
+// date. This will succeed, if the underlying type is either a [float64] or a
+// [json.Number]. Otherwise, nil will be returned.
+func (m MapClaims) ParseNumericDate(key string) *NumericDate {
+	v, ok := m[key]
 	if !ok {
-		return !req
+		return nil
 	}
 
-	switch iat := v.(type) {
+	switch exp := v.(type) {
 	case float64:
-		if iat == 0 {
-			return verifyIat(nil, cmpTime, req)
+		if exp == 0 {
+			return nil
 		}
 
-		return verifyIat(&newNumericDateFromSeconds(iat).Time, cmpTime, req)
+		return newNumericDateFromSeconds(exp)
 	case json.Number:
-		v, _ := iat.Float64()
+		v, _ := exp.Float64()
 
-		return verifyIat(&newNumericDateFromSeconds(v).Time, cmpTime, req)
+		return newNumericDateFromSeconds(v)
 	}
 
-	return false
+	return nil
 }
 
-// VerifyNotBefore compares the nbf claim against cmp (cmp >= nbf).
-// If req is false, it will return true, if nbf is unset.
-func (m MapClaims) VerifyNotBefore(cmp int64, req bool) bool {
-	cmpTime := time.Unix(cmp, 0)
-
-	v, ok := m["nbf"]
-	if !ok {
-		return !req
-	}
-
-	switch nbf := v.(type) {
-	case float64:
-		if nbf == 0 {
-			return verifyNbf(nil, cmpTime, req)
+// ParseClaimsString tries to parse a key in the map claims type as a
+// [ClaimsStrings] type, which can either be a string or an array of string.
+func (m MapClaims) ParseClaimsString(key string) ClaimStrings {
```

Comment:
That is true. I didn't want to change too many things but maybe since we are any way restructuring quite a bit it could be ok. Maybe we can keep a type alias for `ClaimStrings` and deprecate it and remove it a future version.